internal/character/seele/talent.go

```diff
@@ -0,0 +1,73 @@
+package seele
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+)
+
+const (
+	Resurgence  key.Modifier = "seele-resurgence"
+	BuffedState key.Modifier = "seele-buffed-state"
+)
+
+type state struct {
+	c      *char
+	dmgAmt float64
+}
+
+// Enters the buffed state upon defeating an enemy with Basic ATK, Skill, or Ultimate,
+// and receives an extra turn. While in the buffed state,
+// the DMG of Seele's attacks increases by 80% for 1 turn(s).
+// Enemies defeated in the extra turn provided by "Resurgence"
+// will not trigger another "Resurgence."
+
+func init() {
+	modifier.Register(Resurgence, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnTriggerDeath: applyResurgence,
+		},
+	})
+	modifier.Register(BuffedState, modifier.Config{
+		Stacking: modifier.ReplaceBySource,
+	})
+}
+
+// add mod to check for resurgence turns and add extra action
+func (c *char) initTalent() {
+	modState := state{
+		c:      c,
+		dmgAmt: talent[c.info.TalentLevelIndex()],
+	}
+	c.engine.AddModifier(c.id, info.Modifier{
+		Name:   Resurgence,
+		Source: c.id,
+		State:  &modState,
+	})
+}
+
+// enter buffed state and add extra turn if not on resurgence
+func applyResurgence(mod *modifier.Instance, target key.TargetID) {
+	state := mod.State().(*state)
+	// A4 : While Seele is in the buffed state, her Quantum RES PEN increases by 20%.
+	penAmt := 0.0
+	if state.c.info.Traces["102"] {
+		penAmt = 0.2
+	}
+	mod.Engine().AddModifier(mod.Owner(), info.Modifier{
+		Name:   BuffedState,
+		Source: mod.Owner(),
+		Stats: info.PropMap{
+			prop.AllDamagePercent: state.dmgAmt,
+			prop.QuantumPEN:       penAmt,
+		},
+	})
+	// if current turn is resurgence, don't add extra turn. otherwise add extra turn.
+	if state.c.resurgence {
+		state.c.resurgence = false
+	} else {
+		state.c.resurgence = true
+		state.c.engine.InsertAction(state.c.id)
+	}
```

Comment:
Insertion should only happen in the following:
1. Insertion only ever occurs `OnAfterAction`
2. Can only insert iff the action was from Seele (attack, skill, or ult) and triggered a kill
3. If damage was not from any of these cases (such as pursued E6 damage), do not trigger
4. Once a Seele insert starts, can no longer insert new actions until the current turn ends/next turn starts

Reply:
i've been trying to figure this out for a while but about no. 2 though, how do i check for kill trigger inside onAfterAction listener/actionEnd subscriber? can i nest subscriber inside a subscriber like this ? :
```go
c.engine.Events().ActionEnd.Subscribe(func(e event.ActionEnd) {
    c.engine.Events().TargetDeath.Subscribe(func(event event.TargetDeath) {
        // extra turn logics
    } 
}
```

i thought about like using a proc var and set it to true `onAfterAction` and check for it on `TargetDeath` but i don't exactly know the order the events